library/control/src/modules/ProductControl.rb

```diff
@@ -572,13 +572,13 @@ def RequiredFiles(stage, mode)
 
       Builtins.foreach(modules) do |m|
         client = ""
-        if Stage.firstboot
-          client = Ops.get_string(m, "name", "dummy")
+        client = if Stage.firstboot
+          Ops.get_string(m, "name", "dummy")
         else
-          if Builtins.issubstring(Ops.get_string(m, "name", "dummy"), "inst_")
-            client = Ops.get_string(m, "name", "dummy")
+          client = if Builtins.issubstring(Ops.get_string(m, "name", "dummy"), "inst_")
+            Ops.get_string(m, "name", "dummy")
           else
-            client = Ops.add("inst_", Ops.get_string(m, "name", "dummy"))
+            Ops.add("inst_", Ops.get_string(m, "name", "dummy"))
```

Comment:
this demonstrate that check is not nested...double assign to client and above also having empty string initializer looks very bogus for me.